user/apps/user-manage/src/cmd/groupadd.rs

```diff
@@ -0,0 +1,45 @@
+use crate::{
+    check::check::GAddCheck,
+    error::error::{ErrorHandler, ExitStatus},
+    executor::executor::GAddExecutor,
+    parser::parser::GroupParser,
+};
+use libc::geteuid;
+use std::process::exit;
+
+#[path = "../check/mod.rs"]
+mod check;
+#[path = "../error/mod.rs"]
+mod error;
+#[path = "../executor/mod.rs"]
+mod executor;
+#[path = "../parser/mod.rs"]
+mod parser;
+
+#[allow(dead_code)]
+fn main() {
+    let args = std::env::args().collect::<Vec<_>>();
+
+    if unsafe { geteuid() } != 0 {
+        ErrorHandler::error_handle(
+            "permission denied (are you root?)".to_string(),
+            ExitStatus::PermissionDenied,
+        )
+    }
+
+    if args.len() < 2 {
+        ErrorHandler::error_handle(
+            "usage: groupadd [options] groupname".to_string(),
```

Comment:
bin是不是应该显示自己被执行时的文件名而不是固定的程序名？
```suggestion
            format!("usage: {} [options] groupname", args[0]),
```

Reply:
> bin是不是应该显示自己被执行时的文件名而不是固定的程序名？

OK